WIP Confirmation Modals

fhelfer opened the pull request:
![image](https://github.com/user-attachments/assets/8c0fbcc4-a5a4-4ea7-b11c-fb0e0cc4693e)

iszmais committed:
prevent deletion of required file uploads

kergomard committed:
Object: Move CopyWizard to Object

kergomard committed:
Accordion: Deprecate With ILIAS 11

kergomard committed:
Chart: Deprecate With ILIAS 10

kergomard committed:
Excpetions: Move and Deprecate with ILIAS 11

kergomard committed:
Form: Deprecate With ILIAS 12

kergomard committed:
Link: Deprecate With ILIAS 10

kergomard committed:
PermanentLink: Deprecate With ILIAS 11

kergomard committed:
Verification: Deprecate With ILIAS 11

kergomard committed:
YUI: Deprecate With ILIAS 11

kergomard committed:
JQuery: Deprecate With ILIAS 11

kergomard committed:
AssessmentQuestion: Removal

kergomard committed:
QTI: Move To TestQuestionPool

kergomard committed:
CSV: Deprecate With ILIAS 10/11

kergomard committed:
Types: Move Test to BackgroundTasks

kergomard committed:
History: Deprecate With ILIAS 10

kergomard committed:
Utils: Deprecate Utilities with ILIAS 11/12

Utils: Deprecate ilLegacyFormElementsUtil ILIAS 12

Utils: Deprecate ilStr ILIAS 11

Utils: Deprecate ilUtil ILIAS 12

Utils: Deprecate ilShellUtil ILIAS 12

mBeym committed:
Use ks form for saml idp list item add/edit

kergomard committed:
Move Multilingualism to Didactic Template

kergomard committed:
Move ilCache to News

kergomard committed:
Move DataSet to Export

kergomard committed:
Split-Up Advanced Editing

kergomard committed:
Move Block to Container

kergomard committed:
Rename SOAPAuth to AuthSOAP

kergomard committed:
Remove ilCSVUtils and Move Deprecations

kergomard committed:
Move Deprecations of Chart, History & Link to 11

lukas-heinrich committed:
refactor: create test result models to provide typesafe attribute access and ensure semantically correctness

lukas-heinrich committed:
refactor: move result models to namespace ILIAS\Test\Results\Data

lukas-heinrich committed:
refactor: create TestParticipantResultStatus and TestPassResultManager to provide uniform access to test results and optimize queries using in-memory caching

lukas-heinrich committed:
refactor: deprecate the outsourced methods and improve encapsulation

lukas-heinrich committed:
fix: prevent access to non-initialized attributes and array keys

lukas-heinrich committed:
refactor: re-organize typings and code structure

lukas-heinrich committed:
refactor: use apcu cache and use consistent method naming to avoid conflict between 'cache' and 'result_cache'

lukas-heinrich committed:
refactor: use ilias cache service instead of apcu directly

lukas-heinrich committed:
refactor: combine pass result and participant status classes to enhance caching

lukas-heinrich committed:
fix: use test object id instead of test id

lukas-heinrich committed:
refactor: use database constants instead of hard-coded types

lukas-heinrich committed:
refactor: deprecate and replace methods in ilObjTest using TestResultManager

lukas-heinrich committed:
feat: merge proposed adjustments of base test case from #8012

lukas-heinrich committed:
fix: minor fixes, query passed/failed status by mark

lukas-heinrich committed:
test: create test cases for TestResultManager

lukas-heinrich committed:
test: create test cases to ensure that the cache works correctly

lukas-heinrich committed:
fix: prevent the result table from being updated if the current test pass has not been completed

fhelfer committed:
Init ConfirmationModals

fhelfer committed:
Necessary Bugfixes

mjansenDatabay commented on components/ILIAS/UI/src/Component/MessageBox/Exception/LinksAndEntityListConflictException.php:
```diff
@@ -0,0 +1,35 @@
+ * This exception indicates that an UI component was accepted by the JF but is
+ * not backed by a real implementation.
+ */
+class LinksAndEntityListConflictException extends Exception
```
We should use a `DomainException` (as base exception) here.

mjansenDatabay replied:
Maybe we should find a better name. I know it is hard, but maybe `EitherLinksOrEntitiesExpectedException` is already an improvement here.

Another option: _throw new_ `LinksAndEntitiesMustBeExclusiveException`

mjansenDatabay commented on components/ILIAS/UI/src/Implementation/Component/MessageBox/EntityListingRendererFactory.php:
```diff
@@ -0,0 +1,27 @@
+
+use ILIAS\UI\Implementation\Render;
+
+class EntityListingRendererFactory extends Render\DefaultRendererFactory
```
As far as I can see, this renderer is not yet used, correct?

Can you confirm that we need to implement it if we want to render the entity list differently?

fhelfer replied:
A new renderer might not be necessary, we will see
I will delete this for now in the next commit

mjansenDatabay reviewed (changes requested):
Thanks @fhelfer.

I suggest you implement the changers (or have a chat with me, if you have a different opinion). Afterwards, we should schedule a meeting with @oliversamoila to refine the semantic YAML descriptions.

Best regards
Michael

fhelfer committed:
Review Exception renaming and renderer removal

fhelfer committed:
Revert suid

fhelfer committed:
New Renderer for Entities inside MessageBox

lukas-heinrich committed:
fix: Declaration of ILIAS\Cache\Container\Container@anonymous::set(string $key, array|string|int|bool|null $value): void must be compatible with ILIAS\Cache\Container\Container::set(string $key, array|string|int|bool|null $value, ?int $ttl = null): void

fhelfer committed:
Remove Patch trunk

fhelfer committed:
Fix Unit Tests for Feature ConfirmationModals

lukas-heinrich committed:
refactor: improve naming for better consistency

- Use term 'attempt' instead of 'pass'
- Rename 'TestResultManager' and components to 'Repository' and merge with existing repository class
- Adjust naming in DIC
- Use term 'ParticipantResult' instead of 'TestResult'

lukas-heinrich committed:
refactor: remove deprecated methods and static usage of TestResultRepository

lukas-heinrich committed:
fix: failing unit test after renaming methods

kergomard committed:
UI: Do Not Double Encode Values

See: https://mantis.ilias.de/view.php?id=45141

lukas-heinrich committed:
refactor: constructor property promotion and use mark schema to provide mark information in ParticipantResult.php

lukas-heinrich committed:
refactor: rename AttemptResult.php to AttemptSolutions.php and TestPassResult.php to AttemptResult.php to improve the separation between question results and the overall result of an attempt

mjansenDatabay committed:
PermanentLink: Fix perm. link handling if HTTP base path is `/ilias/`

See: https://mantis.ilias.de/view.php?id=44470

# Conflicts:
#	src/StaticURL/Handler/HandlerService.php

matheuszych committed:
UI: respect actual bounds of data in Data Tabl

mjansenDatabay committed:
Mail: Stop sending external emails to expired users

See: https://docu.ilias.de/go/wiki/wpage_8627_1357

ingoj committed:
User: Fix access check to See Role Assignements

See: https://mantis.ilias.de/view.php?id=45314

Access to roleassignments-tab still checked read_users
when using position permissions. Changed to match then
checkWriteRights- functon in this class.

ingoj committed:
User: Adjust Error message on Missing Access

mBeym committed:
ilBadge preloading & caching obj data

schmitz-ilias committed:
MetaData: different fields for LOM digest, more objects for harvester

yvseiler committed:
Änderung Authority to (De-)Assign of CSS

mjansenDatabay committed:
Mail/RoleRecipients: Always generate unique role recipient strings

See: https://mantis.ilias.de/view.php?id=45319
(cherry picked from commit 8725003ccf2fa3bfbf705dc4a2f7fab3811edfd6)

matheuszych committed:
Fixes wrong message being displayed in test launcher button when user used up all possible test passes

mjansenDatabay committed:
Init: Remove super global variable usage

See: https://mantis.ilias.de/view.php?id=45414

schmitz-ilias committed:
Calendar: appointment form to KS

Commit:
Forum: remove outdated db update

lukas-heinrich committed:
refactor: rename toTestResult to avoid misunderstandings

lukastocker committed:
LSO: 45466, adjust function to copy extro and intro content page objects.

https://mantis.ilias.de/view.php?id=45466

mBeym committed:
Remove deprecation from Chart component

mBeym committed:
Remove deprecation from History component

mBeym committed:
Remove deprecation from Link component

mBeym committed:
Remove deprecation from CSV component

mBeym committed:
Remove deprecation from Utilities component

mBeym committed:
Remove deprecation from JQuery component

mBeym committed:
Remove deprecation from YUI component

mBeym committed:
Remove deprecation from Verification component

mBeym committed:
Remove deprecation from PermanentLink component

mBeym committed:
Remove deprecation from Exceptions component

mBeym committed:
Remove deprecation from Accordion component

matheuszych committed:
Fixes issues where messagebox is not displayed, when adding already added participants to booking pool

matheuszych committed:
Fixes issue when creating a schedule and leaving the deadline empty when x hours before time slot begins is selected

matheuszych committed:
Fixes error when trying to delete booking items in a no schedule booking pool

fneumann committed:
[MathJax] reduce service to wrapper

fneumann committed:
[MathJax] fix deprecated calls of ilMathJax

chfsx committed:
[FIX] missing variable in some cases

schmitz-ilias committed:
MetaData: remove required from cp selection in harvester config (45465)

matheuszych committed:
Fixes issues when trying to access additional booking information in specific cases

iszmais committed:
fix typo in dc deletion (#9854)

aaronbidzan committed:
Changes language variable in header message when booking for other participants in booking pool

matheuszych committed:
Fixes issue when trying to get newsfeed url

iszmais committed:
remove actions from the favorites slate (#9857)

matheuszych committed:
Changes language variable in repository action menu from common#:#edit_content to book#:#book_booking_objects

matheuszych committed:
Fixes filtering of participants with no bookings in booking pool

fneumann committed:
[MathJax] Remove old MathJax service functionality

Keep the public interface of ilMathJax
to prevent php errors in other ILIAS components.

A bunch of PRs will replace these calls with the use
of the new Legacy\LatexContent UI component.

The MathJax component will be deleted completely
one all calls are replaced.

fneumann committed:
[MathJax] Remove old MathJax service functionality

Keep the public interface of ilMathJax
to prevent php errors in other ILIAS components.

A bunch of PRs will replace these calls with the use
of the new Legacy\LatexContent UI component.

The MathJax component will be deleted completely
one all calls are replaced.

fneumann committed:
MathJax: Remove entries from settings table

fneumann committed:
MathJax: Adapt README

alex40724 committed:
learning module: fixed array access on copy

fneumann committed:
Language: Add Latex edit note and cleanup

The edit note will be used in different components
as a byline to the textarea input.

All obsolete mathjax related variables are removed

schmitz-ilias committed:
Course: move membership limitation tables to KS

thojou committed:
Test: Fix ManScoringPax Finalized Solutions Exist

See https://mantis.ilias.de/view.php?id=45516

chfsx committed:
[FIX] added testcase for docu

thojou committed:
T&A 45249: add position of question to results

This commit is related to Mantis-Ticket https://mantis.ilias.de/view.php?id=45249.
It ports the changes originally implemented by @nhaagen in PR#9712, commit https://github.com/ILIAS-eLearning/ILIAS/pull/9712/commits/c7074ca4f9fb416845e8a071a8b7c2eefd9133f6
to ILIAS 10 and later versions.

schmitz-ilias committed:
Poll: removed unused fs storage class

schmitz-ilias committed:
Session: removed unused fs storage class

chfsx committed:
[FIX] code tweaks in file and WOPI

lukas-heinrich committed:
fix: code style and failing unit tests

mjansenDatabay committed:
OrgUnits/SOAP: Pull from local DI after initialization

See: https://mantis.ilias.de/view.php?id=45527

fwolf-ilias committed:
Change column order in T&A 'Participants' tab to
"Status of Attempt" → "Total Time Remaining" → "Total Time Available"

kergomard committed:
Object: Fix Setup Status Agent

mjansenDatabay committed:
Forum: Fix obsolete item group header in thread list

See: https://mantis.ilias.de/view.php?id=45532

yvseiler committed:
Maintenance.md: Change CSS Authority for Security Reports and Issues due to personnel changes to catenglaender (who also has the role of auhtority to Sign off Code Changes). (#9905)

qualitus-dahme committed:
Type error fixed (#9909)

* Type error fixed (https://mantis.ilias.de/view.php?id=45429)

qualitus-dahme committed:
Revert PR https://mantis.ilias.de/view.php?id=44862 (#9912)

This reverts commit c49980819e6f307896ef190ccb07f79e76b026b9.

qualitus-dahme committed:
Mantis #45149 - Fix Scorm1.2 favicons again for trunk #9913 (#9914)

* Mantis #45149 - Fix Scorm1.2 favicons again for Ilias9

kergomard committed:
ILIASObject: Fix TitleAndIconVisibility Classname

iszmais committed:
refactor dc table edit form (#9919)

mjansenDatabay committed:
FileDelivery: Fix implicit null issue

mjansenDatabay committed:
Merge remote-tracking branch 'origin/trunk' into trunk

chfsx committed:
[FIX] APCu cache misses and performance optimization

# Conflicts:
#	components/ILIAS/Cache/src/Container/ActiveContainer.php

chfsx committed:
[FIX] lock cache test

fneumann committed:
Setup: Remove MathJax configuration from docu

mjansenDatabay committed:
Saml: Change variable style

mjansenDatabay committed:
User/Accessibility: Fix regression caused by #faa71d1ddfb2e4f09303c86475358eb3e6e2702a

mjansenDatabay committed:
Merge remote-tracking branch 'origin/trunk' into trunk

lscharmer committed:
Chatroom: Replace moment js with native code

nhaagen committed:
[FIX] #43621 UI/Data: increase `UI\URLBuilder` max length, add warning to `Table\Data`. (#9580)

* Fixes https://mantis.ilias.de/view.php?id=43621
* Increase `UI\URLBuilder` max length
* Add warning to `Table\Data` if max length is exceeded
* Disable `Table\Data` selection if max length is exceeded

thibsy committed:
[FIX] UI: `Table\Ordering` and `Table\Data` large example.

chfsx committed:
[FIX] correct URL appendices on WOPI calls

# Conflicts:
#	components/ILIAS/WOPI/classes/Embed/EmbeddedApplication.php

chfsx committed:
[FIX] 0045542: tile images and avatar images are not shown with src/FileDelivery 'delivery_method' => 'xaccel'

chfsx committed:
[FIX] 0045558: File: Usage of ilObjectDIC in \ilObjFileAccess::_lookupOnline results in PHP error

chfsx committed:
[FIX] import issue in BGT setup

chfsx committed:
[FIX] missing new parameter in file input

kergomard committed:
Object: Fix Remaining Steps Count

See: https://mantis.ilias.de/view.php?id=45559

kergomard committed:
Test: Add Byline to Presentation of QuestionTitles

See: https://mantis.ilias.de/view.php?id=45463

kergomard committed:
Test: Add Missing Parameter to Export-Call

See: https://mantis.ilias.de/view.php?id=44608

mjansenDatabay committed:
Test: Fix type check against redirection URL

See: https://mantis.ilias.de/view.php?id=45562

kergomard committed:
Test: Refactor Redirect Function

schmitz-ilias committed:
Calendar: hide recurrence end inputs in recurrence radios

kergomard committed:
Test: Show Other Button-Label on Overview at Start

See: https://mantis.ilias.de/view.php?id=45462

kergomard committed:
Test: Add Better Test for Value as '0' is Valid

See: https://mantis.ilias.de/view.php?id=45479

kergomard committed:
Test: Fix an Option on Last Commit

kergomard committed:
Test: Fix Start-End Switcheroo

See: https://mantis.ilias.de/view.php?id=45537

nhaagen committed:
[FEATURE] UI: add `Field\Numeric::withStepSize()` to support `float` values. (#9676)

* Add `Field\Numeric::withStepSize()` method.
* Add `numeric_with_decimals.php` example.
* Add `step` attribute to numeric HTML inputs.
* Add PHPUnit test cases for `float` values.

kergomard committed:
Object: Fix Issues on Copying Translations

See: https://mantis.ilias.de/view.php?id=45563

kergomard committed:
AccessControl: Fix Implicit null

fhelfer committed:
Fix approve_contact command handling in ilObjUserGUI for contact requests

lscharmer committed:
NPM: Bump linkifyjs and linkify-element from 4.1.3 to 4.3.2

kergomard committed:
Test: Fix Correction of ErrorText-Word-Groups

See: https://mantis.ilias.de/view.php?id=45153

kergomard committed:
AccessControl: Fix Flacking Test

fneumann committed:
Admin: Show Access Denied Instead of Throwing

https://mantis.ilias.de/view.php?id=43449

Cherry-picked from release_10

matheuszych committed:
Test: Fixes Mixed Up Parameters

See: https://mantis.ilias.de/view.php?id=45270

mjansenDatabay committed:
Cron: Optimize weekly schedule checks

QualitusHartwig committed:
fix Mantis #40129 (#9944)

Co-authored-by: Uwe Kohnle <kohnle@internetlehrer-gmbh.de>

mjansenDatabay committed:
Saml: Remove obsolete code in logout process

The code has been moved to `ilSamlAppEventListener`.

(cherry picked from commit 40c83dcb31d646e8df53d800c07b7ecdbcbc61a8)

thibsy committed:
[FIX] #45446 UI: submission of multiple `Field\File` inputs in one form.

* Fixes https://mantis.ilias.de/view.php?id=45446
* Add example to showcase the behaviour of multiple `Field\File` inputs in one form.

kergomard committed:
Test: Fix Rendering of Quotes in Longmenu-Question

See: https://mantis.ilias.de/view.php?id=45139

fneumann committed:
RTE, Utilities: Add function to replace latex span (#9870)

* Rte: Add function to replace a latex span

The span element used by the RTE editor
for latex code is not supported by MathJax.
So it needs to be replaced by the supported
delimiters [tex] ... [/tex].

This function should be called by components
that display RTE content.

* Utilities: Replace MathJax call

Convert the RTE specific delimiters to [tex] ... [/tex]

* RTE: apply suggestion for replaceLatexSpan

fneumann committed:
Forum: Migrate MathJax calls to UI (#9871)

* Forum: Migrate MathJax calls to UI

* Forum: Add return type

* Forum: replace latex span of tiny

lukas-heinrich committed:
refactor: fix variable naming and improve readability

fneumann committed:
T&A: Restore Latex Support (#9916)

Thx @kergomard for review and approve.

mjansenDatabay committed:
Mail: Use `IL_CAL_DATETIME` to present date/time on dashboard mails

See: https://mantis.ilias.de/view.php?id=45587

kergomard committed:
AccessControl: Fix Missing HTTPService

chlulei committed:
Course: ilCourseParticipantsGroupsTableGUI kitchensink implementation

ingoj committed:
User: Hide Filer in Users Roles

Set the filter value to 0 for users that don't have edit access.

fhelfer committed:
Add Configurable Logout Target to Saml Logout event

ingoj committed:
User: Remove Checkbox-Column When No Edit Access

iszmais committed:
fix selection field presentation and copy field on copy (#9956)

kergomard committed:
Test: Tiny Refactoring In QuestionPageGUI

iszmais committed:
fix info screen from list for dc (#9959)

mjansenDatabay committed:
Mail: Allow placeholders within salutation placeholder

This change makes it possible to use `{{FIRST_NAME}}`, `{{LAST_NAME}}` and `{{LOGIN}}` within the
`{{MAIL_SALUTATION}}` placeholder.

See: https://mantis.ilias.de/view.php?id=45607

(cherry picked from commit 37b3dc52f1a006a8d68f4dc0a21df2a3d6984235)

lukas-heinrich committed:
fix: reset TestDIC after adding globals to force Singleton to be recreated

QualitusHartwig committed:
Fix Mantis #40403: Wrong Scorm Version on Container Import (#9963)

matheuszych committed:
Test: Fixe Unexpected Substitution of Results in FormulaQuestion

See: https://mantis.ilias.de/view.php?id=31393

matheuszych committed:
Test: Add Remaining Time to Submission Review

See: https://mantis.ilias.de/view.php?id=45355
and https://mantis.ilias.de/view.php?id=45199

qualitus-dahme committed:
Scorm: Change sequence of parameters when transforming datetime (#9969)

* SCORM Debug Tool without YUI2

* check if array key exists

* Scorm: Change sequence of parameters when transforming datetime (#9825)

* Scorm: Change sequence of parameters when transforming datetime (cherry-picked from release_9: 1111a6eb9e3)

* Possible type error fixed (https://mantis.ilias.de/view.php?id=45306)

* Scorm: Readd Form to Import Tracking Data (#9774)

See: https://mantis.ilias.de/view.php?id=45081

* Type error fixed (https://mantis.ilias.de/view.php?id=45429)

* Revert "Scorm: Change sequence of parameters when transforming datetime (#9826)"

This reverts commit c49980819e6f307896ef190ccb07f79e76b026b9.

* Scorm: Change sequence of parameters when transforming datetime (#9825)

* Scorm: Change sequence of parameters when transforming datetime (cherry-picked from release_9: 1111a6eb9e3)

* Revert "Scorm: Change sequence of parameters when transforming datetime (#9826)"

This reverts commit c49980819e6f307896ef190ccb07f79e76b026b9.

* Mantis #45149 - Fix Scorm1.2 favicons again for Ilias9

* FileDelivery: Fix implicit null issue

* refactor dc table edit form (#9919)

* [FIX] APCu cache misses and performance optimization

# Conflicts:
#	components/ILIAS/Cache/src/Container/ActiveContainer.php

* [FIX] lock cache test

* ilBadge preloading & caching obj data

* Use ks form for saml idp list item add/edit

* Saml: Change variable style

* User/Accessibility: Fix regression caused by #faa71d1ddfb2e4f09303c86475358eb3e6e2702a

* Setup: Remove MathJax configuration from docu

* UI: Do Not Double Encode Values

See: https://mantis.ilias.de/view.php?id=45141

* [FIX] #43621 UI/Data: increase `UI\URLBuilder` max length, add warning to `Table\Data`. (#9580)

* Fixes https://mantis.ilias.de/view.php?id=43621
* Increase `UI\URLBuilder` max length
* Add warning to `Table\Data` if max length is exceeded
* Disable `Table\Data` selection if max length is exceeded

* [FIX] UI: `Table\Ordering` and `Table\Data` large example.

* [FIX] correct URL appendices on WOPI calls

# Conflicts:
#	components/ILIAS/WOPI/classes/Embed/EmbeddedApplication.php

* [FIX] 0045542: tile images and avatar images are not shown with src/FileDelivery 'delivery_method' => 'xaccel'

* Object: Fix Setup Status Agent

* ILIASObject: Fix TitleAndIconVisibility Classname

* [FIX] 0045558: File: Usage of ilObjectDIC in \ilObjFileAccess::_lookupOnline results in PHP error

* [FIX] import issue in BGT setup

* [FIX] missing new parameter in file input

* Object: Fix Remaining Steps Count

See: https://mantis.ilias.de/view.php?id=45559

* Test: Add Byline to Presentation of QuestionTitles

See: https://mantis.ilias.de/view.php?id=45463

* Test: Add Missing Parameter to Export-Call

See: https://mantis.ilias.de/view.php?id=44608

* Change column order in T&A 'Participants' tab to
"Status of Attempt" → "Total Time Remaining" → "Total Time Available"

* Test: Fix type check against redirection URL

See: https://mantis.ilias.de/view.php?id=45562

* Test: Refactor Redirect Function

* Calendar: hide recurrence end inputs in recurrence radios

* Test: Show Other Button-Label on Overview at Start

See: https://mantis.ilias.de/view.php?id=45462

* Test: Add Better Test for Value as '0' is Valid

See: https://mantis.ilias.de/view.php?id=45479

* Test: Fix an Option on Last Commit

* Test: Fix Start-End Switcheroo

See: https://mantis.ilias.de/view.php?id=45537

* [FEATURE] UI: add `Field\Numeric::withStepSize()` to support `float` values. (#9676)

* Add `Field\Numeric::withStepSize()` method.
* Add `numeric_with_decimals.php` example.
* Add `step` attribute to numeric HTML inputs.
* Add PHPUnit test cases for `float` values.

* Object: Fix Issues on Copying Translations

See: https://mantis.ilias.de/view.php?id=45563

* AccessControl: Fix Implicit null

* Fix approve_contact command handling in ilObjUserGUI for contact requests

* NPM: Bump linkifyjs and linkify-element from 4.1.3 to 4.3.2

* Chatroom: Replace moment js with native code

* Test: Fix Correction of ErrorText-Word-Groups

See: https://mantis.ilias.de/view.php?id=45153

* Test: Fix ManScoringPax Finalized Solutions Exist

See https://mantis.ilias.de/view.php?id=45516

* AccessControl: Fix Flacking Test

* Fixes issue when trying to get newsfeed url

* Fixes wrong message being displayed in test launcher button when user used up all possible test passes

* Admin: Show Access Denied Instead of Throwing

https://mantis.ilias.de/view.php?id=43449

Cherry-picked from release_10

* Test: Fixes Mixed Up Parameters

See: https://mantis.ilias.de/view.php?id=45270

* Cron: Optimize weekly schedule checks

* fix Mantis #40129 (#9944)

Co-authored-by: Uwe Kohnle <kohnle@internetlehrer-gmbh.de>

* Saml: Remove obsolete code in logout process

The code has been moved to `ilSamlAppEventListener`.

(cherry picked from commit 40c83dcb31d646e8df53d800c07b7ecdbcbc61a8)

* Test: Fix Rendering of Quotes in Longmenu-Question

See: https://mantis.ilias.de/view.php?id=45139

* RTE, Utilities: Add function to replace latex span (#9870)

* Rte: Add function to replace a latex span

The span element used by the RTE editor
for latex code is not supported by MathJax.
So it needs to be replaced by the supported
delimiters [tex] ... [/tex].

This function should be called by components
that display RTE content.

* Utilities: Replace MathJax call

Convert the RTE specific delimiters to [tex] ... [/tex]

* RTE: apply suggestion for replaceLatexSpan

* Forum: Migrate MathJax calls to UI (#9871)

* Forum: Migrate MathJax calls to UI

* Forum: Add return type

* Forum: replace latex span of tiny

* T&A: Restore Latex Support (#9916)

Thx @kergomard for review and approve.

* Mail: Use `IL_CAL_DATETIME` to present date/time on dashboard mails

See: https://mantis.ilias.de/view.php?id=45587

* AccessControl: Fix Missing HTTPService

* Mail/RoleRecipients: Always generate unique role recipient strings

See: https://mantis.ilias.de/view.php?id=45319
(cherry picked from commit 8725003ccf2fa3bfbf705dc4a2f7fab3811edfd6)

* Add Configurable Logout Target to Saml Logout event

* fix selection field presentation and copy field on copy (#9956)

* Test: Tiny Refactoring In QuestionPageGUI

* prevent deletion of required file uploads

* fix info screen from list for dc (#9959)

* Mail: Allow placeholders within salutation placeholder

This change makes it possible to use `{{FIRST_NAME}}`, `{{LAST_NAME}}` and `{{LOGIN}}` within the
`{{MAIL_SALUTATION}}` placeholder.

See: https://mantis.ilias.de/view.php?id=45607

(cherry picked from commit 37b3dc52f1a006a8d68f4dc0a21df2a3d6984235)

* Fix Mantis #40403: Wrong Scorm Version on Container Import (#9963)

* [FIX] #45446 UI: submission of multiple `Field\File` inputs in one form.

* Fixes https://mantis.ilias.de/view.php?id=45446
* Add example to showcase the behaviour of multiple `Field\File` inputs in one form.

* Test: Fixe Unexpected Substitution of Results in FormulaQuestion

See: https://mantis.ilias.de/view.php?id=31393

* Test: Add Remaining Time to Submission Review

See: https://mantis.ilias.de/view.php?id=45355
and https://mantis.ilias.de/view.php?id=45199

* Scorm: Change sequence of parameters when transforming datetime

---------

Co-authored-by: kergomard <13102171+kergomard@users.noreply.github.com>
Co-authored-by: mjansen <mjansen@databay.de>
Co-authored-by: iszmais <45942348+iszmais@users.noreply.github.com>
Co-authored-by: Fabian Schmid <fabian@sr.solutions>
Co-authored-by: Marvin Beym <mBeym@databay.de>
Co-authored-by: fneumann <fneumann@databay.de>
Co-authored-by: Stephan Kergomard <webmaster@kergomard.ch>
Co-authored-by: Nils Haagen <nhaagen@cat06.de>
Co-authored-by: Thibeau Fuhrer <thibeau@sr.solutions>
Co-authored-by: Fabian Wolf <wolf@ilias.de>
Co-authored-by: Tim Schmitz <schmitz@leifos.de>
Co-authored-by: Fabian Helfer <fhelfer@databay.de>
Co-authored-by: Lukas Scharmer <lscharmer@databay.de>
Co-authored-by: Thomas Joußen <tjoussen@databay.de>
Co-authored-by: Matheus Zych <mzych@databay.de>
Co-authored-by: Matheus Zych <37047867+matheuszych@users.noreply.github.com>
Co-authored-by: QualitusHartwig <hartwig@qualitus.de>
Co-authored-by: Uwe Kohnle <kohnle@internetlehrer-gmbh.de>
Co-authored-by: iszmais <iszmais@databay.de>

kseniialf committed:
Update e-mail-address in  how-to-write-a-privacy.md (#9960)

* Update e-mail-address in  how-to-write-a-privacy.md

* Remove enpty line in how-to-write-a-privacy.md

iszmais committed:
fix translation of memebership manage confirmation (#9973)

selyesa committed:
Update how-to-write-a-privacy.md

Please accept changes. I deleted "via the [ILIAS issue tracker](https://mantis.ilias.de)"

kergomard committed:
Test: Fix Leak of Scoring Information

See: https://mantis.ilias.de/view.php?id=45428

ingoj committed:
User: Fixes After Review

schmitz-ilias committed:
Session/Membership: add mail button label in session membership view (41320)

kergomard committed:
Test: Fix Issues With new Repository

kergomard committed:
User: Remove Roles Tab & Better Redirect

smeyer-ilias committed:
Merge pull request #9951 from leifos-gmbh/11_courseparticipantsgroupstable_kitchensink

Course: ilCourseParticipantsGroupsTableGUI kitchensink implementation

schmitz-ilias committed:
Course: fix pagination in grouping tables

mjansenDatabay committed:
Chat/Mail/Saml: Add parameter to action URL

alex40724 committed:
copage: make code more rubost against missing arrary keys

kergomard committed:
Test: Avoid Crash on Leftover Text Solution

See: https://mantis.ilias.de/view.php?id=45581

kergomard committed:
Test: Fix Wrong Reset of FinalizedBy

See: https://mantis.ilias.de/view.php?id=45599

kergomard committed:
Test: Some Fixes in Results Repository

matthiaskunkel committed:
Fixed Mantis #45528: Incorrect: Available on ‘Info’-tab

(cherry picked from commit c93cbeb484d909ad0888039fafbe01247039a506)

matthiaskunkel committed:
Fixed Mantis #45151: Counterfactual Success Message When Moving Objects Via Clipboard

schmitz-ilias committed:
Conditions: a bunch of wording changes in precondition settings (43249)

schmitz-ilias committed:
Conditions: include lp mode in condition operator labels (43250)

schmitz-ilias committed:
Conditions: a bit of cleanup

kergomard committed:
[FIX] #45613 UI: submission of `Field\File` inputs. (#9979)

* Fixes https://mantis.ilias.de/view.php?id=45613

schmitz-ilias committed:
Merge pull request #9929 from mjansenDatabay/hotfix/9/44249

Tracking: Fix access to undefined array key in CSV export of matrix t…

kergomard committed:
Test: Fix Ghost Attempt on Overview

See: https://mantis.ilias.de/view.php?id=45625

kergomard committed:
Test: Fix Ghost Attempt on Overview

See: https://mantis.ilias.de/view.php?id=45625

mjansenDatabay committed:
Mail: Make parser more robust

lscharmer committed:
Badge: Add CSRF token check for table actions

mjansenDatabay committed:
File/Workspace/WOPI: Fix return URL for workspace files

See: https://mantis.ilias.de/view.php?id=45630

matthiaskunkel committed:
Fixed Mantis #45553: Question Pool for Tests

(cherry picked from commit 58675f1253107af5007e86b78703ace9b22b2959)

c-knof committed:
Fixed Mantis #45608: Wrong timestamp after updating language (#9989)

* Ensure consistent handling of UTC timezones across language-related components.

* Add explicit UTC timezone handling in ilObjLanguageFolderGUI.

fhelfer committed:
Mail: Add render for Placeholder in MAIL_SALUTATION (#9978)

Co-authored-by: Michael Jansen <mjansen@databay.de>

(cherry picked from commit a0301d901b615e816208d3da6be3c2f6e559239a)

rfalkenstein committed:
removed rf from security.md

mBeym committed:
Fix wrong use statement for class under Container/Block

mBeym committed:
Fix wrong path to template under ILIAS/Container

klees committed:
UI: update coordinator-docu (#9851)

The maintainer and coordinator role have been unified in the authority model,
the docu for the UI framework was not adjusted accordingly.

kergomard committed:
User: Fix Retrieval of OwnershipManagementGUI

mjansenDatabay committed:
Forum/TinyMCE: Don't allow aligning/justifying text when answering

See: https://mantis.ilias.de/view.php?id=45638

fhelfer committed:
IL_CAL_UNIX integer cast for DateHelper::formatDate

c-knof committed:
Fixed Mantis #45608: Wrong timestamp after updating language Part 2: changed setup as well (#9993)

Use `gmdate` instead of `date` for consistent UTC timestamps

kergomard committed:
[FIX] #45352: UI: add `Field\Select` translation for required inputs. (#9758)

* Fixes https://mantis.ilias.de/view.php?id=45352
* Add problem builder to required input constraint for proper translation
* Use proper language mock for common field rendering tests.

QualitusHartwig committed:
Fix Mantis #45640: ($iWeight) must be of type float (#9996)

kergomard committed:
DidacticTempl: Fix Implicitely Nullable Parameter

schmitz-ilias committed:
Condition: revert strictness of circle check (45618, 43311)

mjansenDatabay committed:
Certificate: Ensure global date/time constants exist

chlulei committed:
Merge pull request #9983 from leifos-gmbh/10_mantis_ilias_44061

44061: Failed test: Persönliche Daten aus gleicher Version importieren

chlulei committed:
44061: Failed test: Persönliche Daten aus gleicher Version importieren

gvollbach committed:
Foruml: Add parameter to action URL (#9981)

mjansenDatabay committed:
Forum: Add micro optimizations (while spotting some to-dos regarding the tables)

mjansenDatabay committed:
Merge remote-tracking branch 'origin/trunk' into trunk

mjansenDatabay committed:
Auth: Add token for table actions

mjansenDatabay committed:
ResourceStorage: Use more specific name for "View Mode" query parameter

See: https://mantis.ilias.de/view.php?id=45364

# Conflicts:
#	components/ILIAS/ResourceStorage/classes/Resources/class.ilResourceOverviewGUI.php

# Conflicts:
#	components/ILIAS/ResourceStorage/classes/Collections/View/Request.php
#	components/ILIAS/ResourceStorage/classes/Resources/UI/ResourceListingUI.php
#	components/ILIAS/ResourceStorage/classes/Resources/class.ilResourceOverviewGUI.php

gvollbach committed:
Forum: Refactor notification table actions

Co-authored-by: Michael Jansen <mjansen@databay.de>

alex40724 committed:
SystemCheck/Repository/Tree: Fix PHP 8.x issues (#9178)

See: https://mantis.ilias.de/view.php?id=44533

alex40724 committed:
MediaCast/CustomSkins: Make JS code extraction more robust (#9300)

See: https://mantis.ilias.de/view.php?id=43066

alex40724 committed:
droped explicit max size setting

alex40724 committed:
[FIX] 44658: Modules/Survey: Fix cron job 'Remind users to participate' (#9699)

Thanks for the PR!

iszmais committed:
remove glyph (#10010)

chfsx committed:
0045418: Wiederaufnahme der Dateiendung .vtt in die Positivliste der Datei-Services

chfsx committed:
[FIX] 0045488: File goto link redirects authenticated users without access to login (Login Loop)

# Conflicts:
#	components/ILIAS/StaticURL/src/Response/CannotReach.php

# Conflicts:
#	components/ILIAS/StaticURL/src/Context.php

Saaweel committed:
Refactor: replace TYPE_MODULES with TYPE_COMPONENT in question pools and tests

Replaces ilComponentInfo::TYPE_MODULES with ilComponentInfo::TYPE_COMPONENT
in code related to question pools and tests to align with the current structure.

chfsx committed:
[FIX] issues with mp4 in safari

alex40724 committed:
Container: fix rendering issue (#45551) (#9928)

alex40724 committed:
CoPage/MediaObject: Fix editing map areas (#9902)

See: https://mantis.ilias.de/view.php?id=45531

Thanks for the PR!

nhaagen committed:
TA: Introduce Anonymous Scoring

See https://docu.ilias.de/go/wiki/wpage_8202_1357 for details.

iszmais committed:
present values of multi combination selections correctly (#10017)

c-knof committed:
Fixed Mantis #45608: Wrong timestamp after updating language

iszmais committed:
[FIX] UI: (re)compile `templates/default/delos.scss` to include missing changes. (#10003)

* This is a reminder that we should not track compiled files inside our repository - this needs to be a build-step.

iszmais committed:
remove ilTooltip (#10012)

fhelfer committed:
Remove copyright from ConfirmationModal examples

fhelfer committed:
Fix UnitTest ComponentRendererFSLoader

fhelfer committed:
Add missing copyrights
